lib/SILOptimizer/Utils/ConstExpr.cpp

```diff
@@ -553,6 +768,23 @@ ConstExprFunctionState::evaluateFlowSensitive(SILInstruction *inst) {
   if (auto apply = dyn_cast<ApplyInst>(inst))
     return computeCallResult(apply);
 
+  if (isa<StoreInst>(inst)) {
+    auto stored = getConstantValue(inst->getOperand(0));
+    if (!stored.isConstant())
+      return stored;
+
+    return computeFSStore(stored, inst->getOperand(1));
+  }
+
+  // Copy addr is a load + store combination.
+  if (auto *copy = dyn_cast<CopyAddrInst>(inst)) {
```

Comment:
It would be good to add a test for this functionality. This should probably be a SIL-level test so that you can guarantee that the test continues to test the functionality even if SILGen changes its strategy about what is emitted as a CopyAddr.

Reply:
Added a SIL test!